packages/story-editor/src/types/configProvider.ts

```diff
@@ -0,0 +1,306 @@
+    storyId?: number;
+    templateId?: number;
+    optimizedId?: number;
+    cropOriginId?: number;
+    mutedId?: number;
+    posterId?: number;
+    isMuted?: boolean;
+    mediaSource?: string;
+    trimData?: TrimData;
+    baseColor?: string;
+    blurHash?: string;
+    isGif?: boolean;
+    altText?: string;
+  };
+  originalResourceId: ResourceId;
+  resource: Resource;
+};
+
+export type Taxonomy = {
+  name: string;
+  slug: string;
+  capabilities: Record<string, string>;
+  description?: string;
+  labels: Record<string, string>;
+  types: string[];
+  showCloud?: boolean;
+  hierarchical: boolean;
+  restBase: string;
+  restNamespace: string;
+  visibility: Record<string, boolean>;
+  restPath: string;
+};
+
+export interface APICallbacks {
+  addPageTemplate?: (data: TemplateData) => Promise<PageTemplate>;
+  autoSaveById?: (story: Story) => Promise<Story>;
+  createTaxonomyTerm?: (
+    endpoint: string,
+    term: {
+      name: string;
+      parent?: number;
+      slug?: string;
+    }
+  ) => Promise<Term>;
+  deleteMedia?: (id: number) => Promise<boolean>;
+  deletePageTemplate?: (id: number) => Promise<boolean>;
+  getAuthors?: () => Promise<Author[]>;
+  getCurrentUser?: () => Promise<User>;
+  getCustomPageTemplates?: (page: number | boolean) => Promise<{
+    hasMore: boolean;
+    templates: PageTemplate[];
+  }>;
+  getFonts?: (props: GetFontProps) => Promise<Font[]>;
+  getHotlinkInfo?: (link: string) => Promise<HotlinkInfo>;
+  getLinkMetadata?: (link: string) => Promise<LinkMetaData>;
+  getMedia?: (props: {
+    mediaType: string;
+    searchTerm: string;
+    pagingNum: number;
+  }) => Promise<{
+    data: Resource[];
+    headers: Record<string, string>;
+  }>;
+  getMediaById?: (id: number) => Promise<Resource>;
+  // @todo Not certain if correct.
+  getMediaForCorsCheck?: () => Promise<Resource[]>;
+  getMutedMediaById?: (id: number) => Promise<VideoResource>;
+  getOptimizedMediaById?: (id: number) => Promise<Resource>;
+  getPageTemplates?: () => Promise<PageTemplate[]>;
+  getPosterMediaById?: (id: number) => Promise<Resource>;
+  getProducts?: () => Promise<Product[]>;
+  getProxyUrl?: (src: string) => string;
+  getStoryById?: (id: number) => Promise<Story>;
+  getTaxonomies?: () => Promise<Taxonomy[]>;
+  getTaxonomyTerm?: (props: {
+    search?: string;
+    per_page?: number;
+  }) => Promise<Term>;
+  saveStoryById?: (data: Story) => Promise<Story>;
+  updateCurrentUser?: (data: {
+    mediaOptimization?: boolean;
+    onboarding?: Record<string, boolean>;
+    trackingOptin?: boolean;
+  }) => Promise<User>;
+  updateMedia?: (id: number, data: Partial<Resource>) => Promise<Resource>;
+  updatePageTemplate?: (
+    id: number,
+    data: Partial<PageTemplate>
+  ) => Promise<PageTemplate>;
+  uploadMedia?: (files: string[], props: UploadMediaProps) => Promise<Resource>;
+}
+
+export interface PostLock {
+  interval: number;
+  showLockedDialog: boolean;
+}
+
+export interface ConfigState {
```

Comment:
Not sure which of here should be optional and which not. Assigned optional to some of the values that are not used in the `story-editor` package at all and only used by `wp-story-editor`.

Reply:
In #12553 I tried creating separate `DashboardConfig` and `WPDashboardConfig` types. That might be a worthwhile path, maybe with generics if needed.

Ideally `story-editor` is not aware at all of any types from `wp-story-editor`. They're separate.

Reply:
Yep, makes sense. I can just remove the wp-specific types from `ConfigState`  and then in `wp-story-editor`, `WPEditorConfig` or something similar should handle those types instead.